flaml/automl.py

```diff
@@ -518,19 +524,20 @@ def _prepare_data(self,
                     count += rare_count
                 logger.info(
                     f"class {label} augmented from {rare_count} to {count}")
-        if 'sample_weight' in self._state.fit_kwargs:
-            X_train_all, y_train_all, self._state.fit_kwargs[
-                'sample_weight'] = shuffle(
+        if self._split_type != 'time':

```

Comment:
It is better to be explicit than to use negation, as split_type may be set to invalid values. Use this in general:

```python
# Define split types that needs shuffle.
SHUFFLE_SPLIT = ["uniform", ]

#... 

if self._split_type in SHUFFLE_SPLIT_TYPES:
    # do shuffle split
else:
    # do non shuffle split, or nothing
```

Reply:
got it, changes made locally